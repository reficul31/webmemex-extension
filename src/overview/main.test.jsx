/* eslint-env jest */

import React from 'react'
import { Provider } from 'react-redux'
import { shallow } from 'enzyme'

import overview from 'src/overview'
import ReduxDevTools from 'src/dev/redux-devtools-component'

import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

function setupStore(overrides) {
    const overview = Object.assign({
        searchResult: {rows: []},
        currentQueryParams: {
            query: '',
            startDate: undefined,
            endDate: undefined,
        },
        activeQueryParams: undefined,
        waitingForResults: false,
    }, overrides)
    return mockStore({overview})
}

describe('Development App', () => {
    test('should test against the given snapshot', () => {
        const store = setupStore()
        const component = shallow(
            <Provider store={store}>
                <div>
                    <overview.components.Overview grabFocusOnMount />
                    <ReduxDevTools />
                </div>
            </Provider>
        )
        expect(component).toMatchSnapshot()
    })
})
