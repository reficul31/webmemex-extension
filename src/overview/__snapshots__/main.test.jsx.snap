// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Development App should test against the given snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Connect(Overview)
        grabFocusOnMount={true}
    />
    <DevTools />
</div>,
  "nodes": Array [
    <div>
      <Connect(Overview)
            grabFocusOnMount={true}
      />
      <DevTools />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
                Object {
                        "clearActions": [Function],
                        "dispatch": [Function],
                        "getActions": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                      }
        }
>
        <div>
                <Connect(Overview)
                        grabFocusOnMount={true}
                />
                <DevTools />
        </div>
</Provider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <div>
            <Connect(Overview)
                        grabFocusOnMount={true}
            />
            <DevTools />
</div>,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Connect(Overview)
                    grabFocusOnMount={true}
          />
          <DevTools />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <Connect(Overview)
                    grabFocusOnMount={true}
          />
          <DevTools />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
>
    <div>
        <Connect(Overview)
            grabFocusOnMount={true}
        />
        <DevTools />
    </div>
</Provider>,
}
`;
